### PR TITLE
Fix-up DID recovery flow.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: components/identity/identity-service.js [hunk 1]
```diff
@@ -303,6 +303,13 @@ function factory($http) {
       }
       localStorage.setItem(
         STORAGE_KEYS.AUTHENTICATED, JSON.stringify(authenticated));
+
+      // update session
+      var session = service.getSession();
+      if(session) {
```
Comment (dlongley) [inline components/identity/identity-service.js]: Should add this sanity check:

```js
if(session && session.id === id) {
```

**Comment by dlongley** [comment]
One minor comment, then LGTM, merge.

File: components/identity/identity-service.js (hunk 1 above)
Comment (mattcollier) [inline components/identity/identity-service.js]: do you mean:
```
if(session && session.id === authenticated.id) {
```

Comment (dlongley) [inline components/identity/identity-service.js]: No, I'm using the `id` that was passed in for what we're making permanent. If we don't have the same session as what we're making permanent, we shouldn't change its public key ID to the wrong one.

**Pull request merged**